index.js

```diff
@@ -73,11 +73,9 @@ module.exports = options => {
 			});
 
 			if ({}.hasOwnProperty.call(props, 'default')) {
-				if (props.type && props.type === 'boolean' && typeof props.default !== 'boolean') {
-					throw new TypeError(`Expected "${key}" default value to be a boolean`);
-				}
-				if (props.type && props.type === 'string' && typeof props.default !== 'string') {
-					throw new TypeError(`Expected "${key}" default value to be a string`);
+				const defaultType = typeof props.default;
+				if (props.type && defaultType !== props.type) {
```

Comment:
`if (props.type && typeof props.default !== props.type) {` return the following lint error

> Typeof comparisons should be to string literals.

Reply:
I'm not happy with this, sadly I did not found another way to pass the linter.
Disable `requireStringLiterals` in eslint looks like going too far.
What do you think?

Reply:
You can add `// eslint-disable-line requireStringLiterals` next to it and it will suppress this error just for this line.

Reply:
I've disabled valid-typeof rule, for which requireStringLiterals is an option.